packages/selleo-design-core/src/Icon.tsx

```diff
@@ -0,0 +1,193 @@
+import { useState } from 'preact/hooks';
+import render from 'preact-render-to-string';
+import classNames from 'classnames';
+
+import {
+  BookingDeskIcon,
+  BookingRoomIcon,
+  CheckIcon,
+  ClearIcon,
+  DashboardIcon,
+  FacebookIcon,
+  GithubIcon,
+  GoogleIcon,
+  MenuIcon,
+  PlaceholderIcon,
+  PropertiesIcon,
+  SearchIcon,
+  TwitterIcon,
+  UsersIcon,
+} from './icons';
+import { CaretDownIcon } from './icons/CaretDownIcon';
+import { parseToReact } from './parser';
+import { Button } from './Button';
+import { H4 } from './Headers';
+
+type IconProps = {
+  component: h.JSX.Element;
+  name: string;
+};
+
+type IconsProps = {
+  items: IconProps[];
+};
+
+export const iconsData = [
+  {
+    component: <BookingDeskIcon class="h-[24px] w-[24px]" />,
+    name: 'booking-desk-icon',
+  },
+  {
+    component: <BookingRoomIcon class="h-[24px] w-[24px]" />,
+    name: 'booking-room-icon',
+  },
+  {
+    component: <CaretDownIcon class="h-[24px] w-[24px]" />,
+    name: 'caret-down-icon',
+  },
+  {
+    component: <CheckIcon class="h-[24px] w-[24px]" />,
+    name: 'check-icon',
+  },
+  {
+    component: <ClearIcon class="h-[24px] w-[24px]" />,
+    name: 'clear-icon',
+  },
+  {
+    component: <DashboardIcon class="h-[24px] w-[24px]" />,
+    name: 'dashboard-icon',
+  },
+  {
+    component: <FacebookIcon class="h-[24px] w-[24px]" />,
+    name: 'facebook-icon',
+  },
+  {
+    component: <GithubIcon class="h-[24px] w-[24px]" />,
+    name: 'github-icon',
+  },
+  {
+    component: <GoogleIcon class="h-[24px] w-[24px]" />,
+    name: 'google-icon',
+  },
+  {
+    component: <MenuIcon class="h-[24px] w-[24px]" />,
+    name: 'menu-icon',
+  },
+  {
+    component: <PlaceholderIcon class="h-[24px] w-[24px]" />,
+    name: 'placeholder-icon',
+  },
+  {
+    component: <PropertiesIcon class="h-[24px] w-[24px]" />,
+    name: 'properties-icon',
+  },
+  {
+    component: <SearchIcon class="h-[24px] w-[24px]" />,
+    name: 'search-icon',
+  },
+  {
+    component: <TwitterIcon class="h-[24px] w-[24px]" />,
+    name: 'twitter-icon',
+  },
+  {
+    component: <UsersIcon class="h-[24px] w-[24px]" />,
+    name: 'users-icon',
+  },
+];
+
+export function Icon({ component, name }: IconProps) {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const updateClipboard = (code: string) => {
+    navigator.clipboard.writeText(code);
+  };
+
+  const handleCopy = (variant: 'react' | 'html') =>
+    navigator.permissions
+      .query({ name: 'clipboard-write' as any })
+      .then((result) => {
+        if (result.state === 'granted' || result.state === 'prompt') {
+          const Component = component;
+          const HTMLString = render(Component, {
+            pretty: true,
+          });
+
+          const codeToCopy =
+            variant === 'html' ? HTMLString : parseToReact(HTMLString);
+
+          return updateClipboard(codeToCopy);
+        }
+        alert('permission denied');
+      });
```

Comment:
those things should not be in Icon scope, it's modal scope when you display it

Reply:
Do you think that creating for example "IconModal" component and nest the handlers logic there is a good option?

Reply:
Now, the component would look something like that:
```js
export function Icon({ component, name }: IconProps) {
  const [isModalOpen, setIsModalOpen] = useState(false);

  const classes = classNames(
    'flex flex-col justify-center items-center gap-2 rounded border-[1px] border-neutral-600 min-h-[120px] overflow-hidden p-[5px]',
    {
      'cursor-pointer': !isModalOpen,
    }
  );

  return (
    <>
      <div class={classes} onClick={() => setIsModalOpen(true)}>
        {component}
        <p class="whitespace-nowrap text-ellipsis overflow-hidden inline-block text-xs max-w-[98%]">
          {name}
        </p>
        {isModalOpen && (
          <IconModal
            component={component}
            name={name}
            isModalOpen={isModalOpen}
            setIsModalOpen={setIsModalOpen}
          />
        )}
      </div>
    </>
  );
}
```

Reply:
I undestand, it's done